### DISCUSSION TITLE
ETCD log field 'sent-request-id' exceeds uInt64 range

### DISCUSSION THREAD

**Discussion by ankitg-03** [etcd-io/etcd#16488] [Q&A] [answered]
### Bug report criteria

- [X] This bug report is not security related, security issues should be disclosed privately via security@etcd.io.
- [X] This is not a support request, support requests should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [X] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [X] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

Hi,
We are facing an issue in parsing the log field 'sent-request-id'. 
The data type defined by ETCD is uint64. But after a certain time, the value of the 'sent-request-id' field exceeds the uInt64 range, and we face an error while parsing such value. 
We request you to let us know if ETCD can change the data type of 'sent-request-id'  to string in order to handle the above scenario. 

Example:
**'sent-request-id':12752936093112128921**

 {'timestamp':'2023-06-07T17:33:12.218Z','level':'warn','caller':'etcdserver/v3_server.go:840','sent-request-id':12752936093112128921,'retry-timeout':'500ms','ts':'2023-06-07T17:33:12.218Z','@version':'1','msg':'waiting for ReadIndex response took too long, retrying','metadata':{'pod_uid':'5c313bfb-4ec5-46f7-971e-cd4ac8b12767','node_name':'node-1','container_name':'ab','namespace':'ns','pod_name':'abc'}

### What did you expect to happen?

We expect the value of  'sent-request-id' to be in the range of uint64, and not exceed the same. 

### How can we reproduce it (as minimally and precisely as possible)?

Just by performing normal installation and parsing the mentioned JSON field.

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
etcd Version: 3.5.5
Git SHA: 19002cfc6
Go Version: go1.16.15
Go OS/Arch: linux/amd64


$ etcdctl version
etcdctl version: 3.5.5
API version: 3.5

```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

### Relevant log output

```Shell
{'timestamp':'2023-06-07T17:33:12.218Z','level':'warn','caller':'etcdserver/v3_server.go:840','sent-request-id':12752936093112128921,'retry-timeout':'500ms','ts':'2023-06-07T17:33:12.218Z','@version':'1','msg':'waiting for ReadIndex response took too long, retrying','metadata':{'pod_uid':'5c313bfb-4ec5-46f7-971e-cd4ac8b12767','node_name':'node-1','container_name':'ab','namespace':'ns','pod_name':'abc'}
```

**Comment by jmhbnz** [comment, marked as answer]
Hi @ankitg-03 - Thanks for raising this question.

I might be missing something but `uint64` is the set of all unsigned 64-bit integers (`0` to `18446744073709551615`). Refer to https://go.dev/ref/spec#Numeric_types.

The example you have provided, `12752936093112128921` is ~5 quintillion less than `18446744073709551615`?
